docs/howto/how-to-make-tensorflow-plugin-with-ui/tensorflow-plugin-with-ui.md

```diff
@@ -0,0 +1,1488 @@
+
+#### Function (Update): `__init__`
+
+```python
+    def __init__(self, provider: "AMPProviderV2") -> None:
+        """Construct a plugin."""
+        # snip... 
+        self.model_save_loc = self.provider.tool_config["modelConfig"]["modelName"]
+        self.exported_model = f"{self.model_save_loc}-exported"
+
+        self.provider.io.info("Plugin initialized.")
+```
+
+### Function (Update): `on_complete`
+
+```python
+    def on_complete(self) -> None:
+        """
+        Clean up any plugin resources, or push records for an input tool.
+
+        This method gets called when all other plugin processing is complete.
+In this method, a Plugin designer should perform any cleanup for their plugin.
+        However, if the plugin is an input-type tool (it has no incoming connections),
+        processing (record generation) should occur here.
+
+        Note: A tool with an optional input anchor and no incoming connections should
+        also write any records to output anchors here.
+        """
+        try:
+            if self.MODE == 'DATA':
+                self.setup_data()
+            elif self.MODE == 'TRAIN':
+                fn_args = self.get_model_args()
+                # TODO: Update this to its own function.
+                # NOTE: Need to remove test data pushes
+                df = pd.DataFrame({"Results": [0.0, 0.0]})
+                batch_to_send = pa.RecordBatch.from_pandas(df=df, preserve_index=False)
+                self.provider.write_to_anchor("Output", batch_to_send)
+                q = Queue()
+                p = Process(target=self.create_and_save_model, args=(q, *fn_args))
+                p.start()
+                history = q.get()
+                p.join()
+                history_dict = history.history
+                conf = self.provider.full_config
+                self.info("Setting new history")
+
+                # Pot, meet kettle. Need another method or opportunity for Additional Execercise section?
+                ui_history = {}
+
+                ui_history["trainingLoss"] = history_dict["loss"]
+                ui_history["trainingBinaryAccuracy"] = history_dict["binary_accuracy"]
+                ui_history["validationLoss"] = history_dict["val_loss"]
+                ui_history["validationBinaryAccuracy"] = history_dict["val_binary_accuracy"]
+                conf['Configuration']["modelEvaluation"]['history'] = ui_history
+                self.provider.save_full_config(conf)
+            elif self.MODE == 'EVAL':
+                model = tf.keras.saving.load_model(self.provider.tool_config["modelConfig"]["modelName"], custom_objects={'custom_standardization': custom_standardization})
+            elif self.MODE == 'PREVIEW':
+                self.send_preview_data()
+```
+
+
+The majority of the above speaks for itself at this point in the guide.
+Our most notable (conceptual) addition is likely the `MultiProcessing` section:  
```

Comment:
TW review:

The majority of the above speaks for itself at this point in the guide. The most notable (conceptual) addition is likely the `MultiProcessing` section: